### PR TITLE
Grantham's substitution matrix

### PR DESCRIPTION
**Pull request by KaparaNewbie**
# Add support for R. Grantham's substitution matrix from his 1974 paper

## Types of changes

* [x] :sparkles: New feature (A non-breaking change which adds functionality).
* [ ] :bug: Bug fix (A non-breaking change, which fixes an issue).
* [ ] :boom: Breaking change (fix or feature that would cause existing functionality to change).

## :clipboard: Additional detail

- If you have implemented new features or behaviour
  - **Description of the addition**
  This PR adds support for R. Grantham's substitution matrix from his [1974 paper](https://www.science.org/doi/10.1126/science.185.4154.862?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub%20%200pubmed) in which the distance between any two amino acids is based on three physicochemical properties: composition, polarity, and molecular volume.  
In addition, `default_match=0, default_mismatch=0` are kwargs in the `load_submat` and `parse_ncbi_submat` functions. Existing calls to PAM and BLOSUM aren't changed as a result. For the new `GRANTHAM1974` matrix, I chose `default_mismatch = 100` because 100 is the mean value in Grantham's original table.

  - **Justification of the addition**
  Two justifications should be provided: (a) for the addition of the matrix, (b) and the (unbreaking!) changes to the `load_submat` and `parse_ncbi_submat` functions.
      1. While the existing PAM and BLOSUM matrices describe changes between amino acids from an evolutionary perspective, Grantham's matrix provides a more direct measurement. This measurement, supplemented by a cutoff, may help to discern between [radical and conservative amino acid replacements](https://judgestarling.tumblr.com/post/125812920996/radical-and-conservative-amino-acid).
      2. As Grantham only dealt with the basic 20 amino acids, I needed to support a different `default_mismatch` compared to `default_mismatch == 0` as in the original return value `SubstitutionMatrix(scores, default_match=0, default_mismatch=0)` in the final line of the `parse_ncbi_submat` function. 

  - **A runnable example**
    ```
    julia> using BioAlignments
    julia> sm = GRANTHAM1974
    ```

## :ballot_box_with_check: Checklist

- [X] :art: The changes implemented is consistent with the [julia style guide](https://docs.julialang.org/en/stable/manual/style-guide/).
- [x] :blue_book: I have updated and added relevant docstrings, in a manner consistent with the [documentation styleguide](https://docs.julialang.org/en/stable/manual/documentation/).
- [x] :blue_book: I have added or updated relevant user and developer manuals/documentation in `docs/src/`.
- [ ] :ok: There are unit tests that cover the code changes I have made.
- [ ] :ok: The unit tests cover my code changes AND they pass.
- [ ] :pencil: I have added an entry to the `[UNRELEASED]` section of the manually curated `CHANGELOG.md` file for this repository.
- [x] :ok: All changes should be compatible with the latest stable version of Julia.
- [x] :thought_balloon: I have commented liberally for any complex pieces of internal code.

### CODE REVIEW

**Comment by KaparaNewbie** [comment]
@MillironX, could you kindly review this?

**Review by MillironX — CHANGES_REQUESTED** [review]
I love this addition. I'd never heard of the Grantham matrix, but I think it's worthy of being a part of BioAlignments. I have a few points to make this code blend in seamlessly with the rest of BioJulia

- [ ] Let's not change .gitignore for VSCode, even if I know most of us use it
- [ ] src/submat.jl got a lot of auto-formatted code in it that has nothing to do with adding the Grantham matrix: let's undo those changes
- [ ] I would really appreciate some tests to make sure the matrix is loading correctly, see https://github.com/BioJulia/BioAlignments.jl/blob/9a6a37defeef5176ae82332b67007a4f1ab8784d/test/runtests.jl#L410-L414 for an example of how the BLOSUM matrix is tested
- [ ] Go ahead and add this to the CHANGELOG

You may find [VSCode's source control gutters](https://youtu.be/CKQmerGZJUA) and [interactive rebase](https://docs.gitlab.com/ee/topics/git/git_rebase.html#interactive-rebase) helpful.

Great addition, with awesome documentation. Thanks for contributing!

**Comment by KaparaNewbie** [comment]
Thanks, @MillironX, for your very kind response.
I think I addressed all your points (hopefully there are enough tests).

File: .gitignore [hunk 1]
```diff
@@ -23,4 +23,4 @@ docs/site/
 # environment.
 Manifest.toml
 
-.DS_Store
+.DS_Store
```
[the remark this answers is not in the document]
Comment (MillironX) [reply .gitignore:26]: Once again, potentially useful, but out-of-scope of this PR

**Comment by KaparaNewbie** [comment]
Sounds good, thanks.
Sorry about the `.gitignore` thing. My git skill is (still) embarrassingly noobish.

**Review by kescobo — APPROVED** [review]
Looks good to me - I can't really imagine how it would break anything, and tests pass so :shrug:

**Pull request merged**